hoss/coordinate_utilities.py

```diff
@@ -162,59 +165,234 @@ def get_coordinate_array(
     return coordinate_array
 
 
-def get_1D_dim_array_data_from_dimvalues(
-    dim_values: np.ndarray, dim_indices: np.ndarray, dim_size: int
+def get_1d_dim_array_data_from_dimvalues(
+    dim_values: np.ndarray,
+    dim_indices: np.ndarray,
+    dim_size: int,
 ) -> np.ndarray:
     """
-    return a full dimension data array based on the 2 projected points and
-    grid size
+    return a full dimension data array based upon 2 valid projected values
+    (x and y) given in dim_values which are within the indices given in
+    dim_indices and the full dimension size provided in dim_size. The
+    dim_indices need to be between 0 and less than the dim_size.
+    returns a 1D array of size = dim_size with proper dimension array values
     """
```

Comment:
```
Return a full dimension data array based upon 2 valid projected values
given in dim_values and located by dim_indices. The dim_indices need 
to be between 0 and dim_size. Returns a 1D array of size = dim_size 
with proper dimension array values, with linear interpolation between 
the given dim_values.
```
Sorry for the rewrite, but on this review it is clear this didn't make any sense.

Reply:
UPDrw - https://github.com/nasa/harmony-opendap-subsetter/pull/18/commits/1e5307a1e2b821a834fa6080d74b80cafa935350